authlib/jose/rfc7516/models.py

```diff
@@ -15,10 +15,22 @@ class JWEAlgorithm(object):
     def prepare_key(self, raw_data):
         raise NotImplementedError
 
-    def wrap(self, enc_alg, headers, key):
+    def wrap(self, enc_alg, headers, key, sender_key):
```

Comment:
How about rewrite `wrap` and `unwrap` in `JWEAlgorithmWithTagAwareKeyAgreement`?

In this case, you don't have to change code in `authlib/jose/rfc7518/jwe_algs.py`.

Reply:
I have refactored `JWEAlgorithm` and related interfaces to keep API of the existing algorithms unchanged.

Thank you for the quick feedback!